Microsoft.Toolkit.Uwp.UI.Controls/ImageCropper/ImageCropper.Extensions.cs

```diff
@@ -0,0 +1,288 @@
+                    (int)encoder.BitmapTransform.Bounds.Height);
+                croppedBitmap.SetSource(memoryRandom);
+            }
+
+            return croppedBitmap;
+        }
+
+        /// <summary>
+        /// Save the cropped image to a file.
+        /// </summary>
+        /// <param name="writeableBitmap">The source image.</param>
+        /// <param name="imageFile">The target file.</param>
+        /// <param name="encoderId">The encoderId of BitmapEncoder</param>
+        /// <returns>Task</returns>
+        internal static async Task RenderToFile(WriteableBitmap writeableBitmap, StorageFile imageFile, Guid encoderId)
+        {
+            using (var stream = await imageFile.OpenAsync(FileAccessMode.ReadWrite, StorageOpenOptions.None))
+            {
+                var encoder = await BitmapEncoder.CreateAsync(encoderId, stream);
+                var pixelStream = writeableBitmap.PixelBuffer.AsStream();
+                var pixels = new byte[pixelStream.Length];
+                await pixelStream.ReadAsync(pixels, 0, pixels.Length);
+                encoder.SetPixelData(
+                    BitmapPixelFormat.Bgra8,
+                    BitmapAlphaMode.Premultiplied,
+                    (uint)writeableBitmap.PixelWidth,
+                    (uint)writeableBitmap.PixelHeight,
+                    96.0,
+                    96.0,
+                    pixels);
+                await encoder.FlushAsync();
+            }
+        }
+
+        /// <summary>
+        /// Gets the closest point in the rectangle to a given point.
+        /// </summary>
+        /// <param name="targetRect">The rectangle.</param>
+        /// <param name="point">The test point.</param>
+        /// <returns>A point within a rectangle.</returns>
+        internal static Point GetSafePoint(Rect targetRect, Point point)
+        {
+            var safePoint = new Point(point.X, point.Y);
+            if (safePoint.X < targetRect.X)
+            {
+                safePoint.X = targetRect.X;
+            }
+
+            if (safePoint.X > targetRect.X + targetRect.Width)
+            {
+                safePoint.X = targetRect.X + targetRect.Width;
+            }
+
+            if (safePoint.Y < targetRect.Y)
+            {
+                safePoint.Y = targetRect.Y;
+            }
+
+            if (safePoint.Y > targetRect.Y + targetRect.Height)
+            {
+                safePoint.Y = targetRect.Y + targetRect.Height;
+            }
+
+            return safePoint;
+        }
+
+        /// <summary>
+        /// Test whether the point is in the rectangle.
+        /// Similar to the Rect.Contains method, this method adds redundancy.
+        /// </summary>
+        /// <param name="targetRect">the rectangle.</param>
+        /// <param name="point">The test point.</param>
+        /// <returns>bool</returns>
+        internal static bool IsSafePoint(Rect targetRect, Point point)
+        {
+            if (point.X - targetRect.X < -0.001)
```

Comment:
Math.Abs(point.X - targetRect.X)<0.001

Reply:
This is wrong. When the point is in the target rectangle, point X will be greater than 0.001 in most cases, so that your judgment will return the wrong result.